packages/atomic/src/components/facets/atomic-segmented-facet-scrollable/atomic-segmented-facet-scrollable.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Listen,
+} from '@stencil/core';
+import ArrowRightIcon from 'coveo-styleguide/resources/icons/svg/arrow-right-rounded.svg';
+import ArrowLeftIcon from 'coveo-styleguide/resources/icons/svg/arrow-left-rounded.svg';
+
+@Component({
+  tag: 'atomic-segmented-facet-scrollable',
+  styleUrl: 'atomic-segmented-facet-scrollable.pcss',
+  shadow: true,
+})
+export class AtomicSegmentedFacetScrollable {
+  @Element() private host!: HTMLElement;
+  @Event({bubbles: false}) scrollLeft!: EventEmitter;
+  @Event({bubbles: false}) scrollRight!: EventEmitter;
+
+  @Listen('scrollLeft')
+  @Listen('scrollRight')
+  private slide(direction: Event) {
+    const container = this.host.shadowRoot?.getElementById('horizontalScroll');
```

Comment:
Getting the element from the shadow root every time we try to scroll is inefficient. I tried to save it as a global variable, but it keeped on getting dereferenced for some reason.